### PR TITLE
tests: SQLiteJournal 'Too many variables' bug test added [Ref #25]

### LINKED ISSUE #25: SQLiteJournal: General error: 1 too many SQL variables
I've switched our project to Nette\Caching v2.4 due to another bug (issue #18).

But today, an application has failed with `PDOException #HY000 SQLSTATE[HY000]: General error: 1 too many SQL variables`.

I can't provide error log in RAW HTML output (proprietaly real application), but I can part of Stack Trace as image:

6sjstvs

### PR DESCRIPTION
**Pull request by hranicka**
Fixes #25 with test case for the bug.

### CODE REVIEW

File: tests/Caching/SQLiteJournal.phpt [hunk 1]
```diff
@@ -21,7 +21,7 @@ class SQLiteJournalTest extends IJournalTestCase
 
 	public function createJournal()
 	{
-		return new SQLiteJournal;
+		return new SQLiteJournal(TEMP_DIR . '/sqlitejournal_' . uniqid() . '.sqlite');
```
Comment (hranicka) [inline tests/Caching/SQLiteJournal.phpt]: For "in memory" SQLite storage there were problems :o 

See https://travis-ci.org/nette/caching/builds/71373010


It looks like that test cases shared the same database but it was OK before my changes, I don't know why...


---

Now, it is OK: https://travis-ci.org/nette/caching/builds/71373257

File: src/Caching/Storages/SQLiteJournal.php [hunk 2]
```diff
@@ -108,25 +108,26 @@ public function clean(array $conditions)
 		}
 
 		if (!empty($conditions[Cache::PRIORITY])) {
+			$priority = (int) $conditions[Cache::PRIORITY];
```
Comment (dg) [inline src/Caching/Storages/SQLiteJournal.php]: Why this new variable?

Comment (hranicka) [inline src/Caching/Storages/SQLiteJournal.php]: Sorry, I'll remove it... It's trace after 239a9f6a

**Comment by dg** [comment]
Can you squash it to single commit?

**Comment by hranicka** [comment]
I've cleaned it up.

Really should I squash these commits?
* 6f863365 adds test case for the bug
* 2178eb3 fixes the bug #25
* 080568d fixes a problem with previous SQLiteJournalTest :memory:

**Comment by dg** [comment]
This is fine, thanks!

File: tests/Caching/SQLiteJournal.phpt [hunk 3]
```diff
@@ -21,7 +21,8 @@ class SQLiteJournalTest extends IJournalTestCase
 
 	public function createJournal()
 	{
-		return new SQLiteJournal;
+		static $id = 0;
+		return new SQLiteJournal(TEMP_DIR . '/sqlitejournal_' . ++$id . '.sqlite');
```
Comment (hranicka) [inline tests/Caching/SQLiteJournal.phpt]: It's not possible share :memory: SQLite nor after these changes:

https://github.com/nette/caching/pull/27/files#diff-88c1903cdb7494caec03523cf64621b1R121

Nor if I replace `BEGIN IMMEDIATE` with `BEGIN EXCLUSIVE`.

**Pull request merged**